### ISSUE TITLE
Returners output use "Highstate Formatting"

### ISSUE THREAD

**Issue by evan-ty** [saltstack/salt#36570]
### Description of Issue/Question

Using the Slack returner via:
`sudo salt 'two' test.versions_report --return slack --output=highstate`
returns the output as highstate, but only on the command line.  The output in the slack channel remains unchanged as this:

![image](https://i.imgur.com/MF9AB7Q.png)

My question is, how can I get it to post to slack in a format where it's easy on the eyes?
### Setup

(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

I have only really used the
"user: saltuser"
"color: True"
"file_roots: base: - /srv/salt"

and:

slack.channel: "#saltbuilds"
slack.api_key: "nuked"
slack.username: "me"
slack.as_user: "true"
### Steps to Reproduce Issue

`sudo salt 'two' test.versions_report --return slack --output=highstate`
### Versions Report

(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

`Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
          gitdb: 0.6.4
      gitpython: 2.0.5
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12+ (default, Sep  1 2016, 20:27:38)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.1.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: debian stretch/sid 
        machine: x86_64
        release: 4.3.0-1-amd64
         system: Linux
        version: debian stretch/sid`

Both minion and master are on the same version.

**Comment by cachedout** [comment]
@garethgreenaway Any chance you could peek at this?

**Comment by garethgreenaway** [comment]
@cachedout Will take a look.

**Comment by garethgreenaway** [comment]
@evan-ty Try adding the slack.format_yaml option to the minion configuration.  That gives the output a bit more readable format.

**Comment by evan-ty** [comment]
Apologies for the delay in reply:

I have placed this in my minion:
```
ext_job_cache: slack
slack.channel: "#saltbuilds"
slack.api_key: "nuked"
slack.username: "helloworld"
slack.as_user: "true"
slack.format_yaml: "true"
```

I have restarted the minion: `service salt-minion restart`.  To my dismay, the output is still the same as the one in my original post up top.

Could it be that I'm doing it wrong?  Or rather, it does something else?
Reading the doc:

> slack.yaml_format(optional, format the json in yaml format)

It says it formats the json in yaml, in the Slack output, it does not look like json.
